### ISSUE TITLE
bug(timeseries): GPU utilization 0% during TrainWindowed for ALL backends (v1.20.2)

### ISSUE THREAD

**Issue by dndungu** [zerfoo/zerfoo#207]
## Problem

Despite #166 and #172 being closed, `nvidia-smi` shows **0% GPU utilization** during TrainWindowed for ALL backends on v1.20.2:

```
PatchTST GPU: 0%
TTM GPU:      0%
iTransformer: 1%
```

GPU engine initializes successfully but the actual forward/backward computation runs on CPU.

## Evidence

PatchTST with WOLF_GPU=1 vs without: 13s vs ~60s on 3K rows. The ~5x "speedup" is NOT from GPU compute (0% utilization). It may be from engine initialization caching or different code paths.

TTM: 50s GPU vs 48s CPU — zero benefit (0% utilization).
iTransformer: 58s GPU vs ~60s CPU — zero benefit (1% utilization).

## Root Cause

`TrainWindowed` likely still uses `forwardF64` (pure Go) for the forward pass. The GPU engine may only be used for:
- Weight initialization
- AdamW optimizer updates
- Normalization

The compute-intensive matmul/attention operations are NOT dispatched to GPU.

## Impact

No DL backend can train efficiently on 28K+ row datasets. Only LightGBM and DLinear produce results within practical timeouts. PatchTST, iTransformer, TTM, CfC, FreTS are all blocked.

## Environment

- Zerfoo v1.20.2-dev, ztensor v0.4.1 (v0.5.0)
- DGX Spark GB10 sm_121, CUDA 13.0
- Kernels at /opt/zerfoo/lib/libkernels.so (compiled, loaded)
- All backends tested: patchtst, ttm, itransformer
